packages/components/icon/src/Icon.tsx

```diff
@@ -0,0 +1,135 @@
+import { css, cx } from 'emotion';
+import React, { forwardRef } from 'react';
+import tokens from '@contentful/f36-tokens';
+import { getSpacingStyles, Primitive } from '@contentful/f36-core';
+import type {
+  CommonProps,
+  ComponentVariant,
+  MarginProps,
+  PaddingProps,
+  Simplify,
+} from '@contentful/f36-core';
+import type {
+  ComponentType,
+  ExoticComponent,
+  ReactElement,
+  SVGAttributes,
+} from 'react';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type IconComponent = ExoticComponent<any> | ComponentType<any>;
+
+export type IconSize = 'large' | 'medium' | 'small' | 'tiny';
+
+export type IconVariant = Simplify<ComponentVariant | 'muted' | 'white'>;
+
+const sizes: { [key in IconSize]: { [key in 'height' | 'width']: string } } = {
+  large: {
+    height: '32px',
+    width: '32px',
+  },
+  medium: {
+    height: '24px',
+    width: '24px',
+  },
+  small: {
+    height: '18px',
+    width: '18px',
+  },
+  tiny: {
+    height: '16px',
+    width: '16px',
+  },
+};
+
+const fills: { [key in IconVariant]: string } = {
+  muted: tokens.colorTextLight,
+  negative: tokens.colorNegative,
+  positive: tokens.colorPositive,
+  primary: tokens.colorBlueBase,
+  secondary: tokens.colorContrastMid,
+  warning: tokens.colorWarning,
+  white: tokens.colorWhite,
+};
```

Comment:
Couldn't this be an enum? 🤔

Reply:
It think it could but do you think that would make it easier to use? This is for internal use.

Reply:
I guess it's a tomAYto/tomAHto situation 👌 

Nevermind then

Reply:
I like enums but I dropped them because we're not using them anywhere else. I don't want to rock the boat! 😅